### PR TITLE
Added baseline LSTM (from fb) 

### PR DESCRIPTION
**Pull request by HershGupta25**
Had to update various import statements in our and fb util files to get the water flowing and then added in baseline LSTM. Saved first runs are in model_saves for S2S and STT (STT took ~3 mins for one epoch on my mac air).

### CODE REVIEW

File: utils/model.py [hunk 1]
```diff
@@ -53,12 +58,12 @@ def get_model(args, datasets, device):
         pass
     elif args.model == "RNN_a":
         pass
-    elif args.model == "LSTM":
-        pass
+    elif args.model == ModelEnum.LSTM_SEQ2SEQ.value:
+        enc = LSTMEncoder(input_dim=raw_dim, hidden_dim=hidden_dim, num_layers=nlayers).to(device)
+        dec = LSTMDecoder(input_dim=raw_dim, hidden_dim=hidden_dim, output_dim=raw_dim,device='cpu').to(device)
```
Comment (jrevansjr) [inline utils/model.py:63]: CPU specified as device here, shouldn't we specify device as device?

File: models/Decoders.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+
+class LSTMDecoderWithAttention(LSTMDecoder):
+    def __init__(
+        self,
+        input_dim,
+        output_dim,
+        max_source_length,
+        hidden_dim=128,
+        device="cuda",
```
Comment (jrevansjr) [inline models/Decoders.py:125]: device should be device and not "cuda", have you tested the modules to make sure that they pass? Because the Super takes device as an argument they will end up on different devices if you pass "CPU"

File: main.py [hunk 3]
```diff
@@ -58,22 +62,23 @@ def main():
...
 
         # Compute epoch losses
         epoch_loss = epoch_loss / num_training_sequences
         training_losses.append(epoch_loss)
         epoch_val_loss, val_iter = compute_validation_loss(args, model, datasets, criterion, device, mask)
         validation_losses.append(epoch_val_loss / ((val_iter + 1) * batch_size))
+        print(f"Epoch {epoch} | ")
         print(f"Training loss {epoch_loss} | ")
-        print(f"Validation loss {epoch_val_loss} | ")
+        print(f"Validation loss {epoch_val_loss / ((val_iter + 1) * batch_size)} | ")
```
Comment (jrevansjr) [inline main.py:74]: Can we get language for validation loss and training loss to match, fundamentally they should be calculated in similar ways. val_iter + 1 may equal num_training_sequences, but would be nice to just make sure this is accurate.

**Review by jrevansjr — CHANGES_REQUESTED** [review]
Can we delete the test files (models, configs, etc. prior to pushing), make sure all modules that are added are tested (eg. LSTM with attention), few other comments.

File: main.py (hunk 3 above)
Comment (moduIo) [reply main.py:74]: 100% agree, we need them to be computed in the exact same way
